cmd/preview/deploy.go

```diff
@@ -0,0 +1,228 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package preview
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/analytics"
+	"github.com/okteto/okteto/pkg/cmd/login"
+	"github.com/okteto/okteto/pkg/errors"
+	okErrors "github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/spf13/cobra"
+)
+
+// Deploy Deploy a preview environment
+func Deploy(ctx context.Context) *cobra.Command {
+	var branch string
+	var filename string
+	var repository string
+	var scope string
+	var sourceUrl string
+	var timeout time.Duration
+	var variables []string
+	var wait bool
+
+	cmd := &cobra.Command{
+		Use:   "deploy <name>",
+		Short: "Deploy a preview environment",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := login.WithEnvVarIfAvailable(ctx); err != nil {
+				return err
+			}
+
+			if !okteto.IsAuthenticated() {
+				return okErrors.ErrNotLogged
+			}
+
+			if err := validatePreviewType(scope); err != nil {
+				return err
+			}
+
+			var err error
+			repository, err = getRepository(ctx, repository)
+			if err != nil {
+				return err
+			}
+			branch, err = getBranch(ctx, branch)
+			if err != nil {
+				return err
+			}
+
+			name := ""
+			if len(args) > 0 {
+				name = args[0]
+			}
+
+			varList := []okteto.Variable{}
+			for _, v := range variables {
+				kv := strings.SplitN(v, "=", 2)
+				if len(kv) != 2 {
+					return fmt.Errorf("invalid variable value '%s': must follow KEY=VALUE format", v)
+				}
+				varList = append(varList, okteto.Variable{
+					Name:  kv[0],
+					Value: kv[1],
+				})
+			}
+
+			name, err = executeDeployPreview(ctx, name, scope, repository, branch, sourceUrl, filename, varList, wait, timeout)
+			analytics.TrackCreatePreview(err == nil)
+			if err != nil {
+				return err
+			}
+
+			if wait {
+				log.Success("Preview environment '%s' successfully deployed", name)
+			} else {
+				log.Success("Preview environment '%s' scheduled for deployment", name)
+			}
+			return err
+		},
+	}
+	cmd.Flags().StringVarP(&branch, "branch", "b", "", "the branch to deploy (defaults to the current branch)")
+	cmd.Flags().StringVarP(&filename, "filename", "f", "", "relative path within the repository to the manifest file (default to okteto-pipeline.yaml or .okteto/okteto-pipeline.yaml)")
+	cmd.Flags().StringVarP(&repository, "repository", "r", "", "the repository to deploy (defaults to the current repository)")
+	cmd.Flags().StringVarP(&scope, "scope", "s", "personal", "the scope of preview environment to create. Accepted values are ['personal', 'global']")
+	cmd.Flags().StringVarP(&sourceUrl, "sourceUrl", "", "", "the pull request url to notify.")
```

Comment:
"pull request" is a github concept, in gitlab they are called 'merge requests'. not sure what they are called in other systems. 

`sourceURL` sounds odd, is this what the URL is called in other preview environment (or similar systems)? Also, will this work for any git repository or just github?

Reply:
We chose `sourceURL` exactly because of the name for this in different systems. The idea is the URL pointing to whatever originated the Preview Environment (either a pull request, a merge request, etc.). It doesn't seem to be a generic/standard way of calling this. Do you have a better suggestion?

As for the explanation, I agree we should remove "pull request" or mention the other in the description.

Reply:
Maybe something like this?:

```suggestion
	cmd.Flags().StringVarP(&sourceUrl, "sourceUrl", "", "", "the URL of the original pull/merge request")
```

Reply:
@rlamana  what do we use the URL for? Is it just to show the link to the PR in the UI, or is there some other action that we take?